batch.go

```diff
@@ -120,8 +120,11 @@ func (batch *Batch) Read(b []byte) (int, error) {
 				return size, nil
 			}
 			n = nbytes // return value
+			if nbytes > cap(b) {
```

Comment:
I think it will read more clearly if we move this check down into the if block for the `len` check and to re-slice only in the case where `len < cap`

Reply:
Do you mean:

```
if nbytes > len(b) {
  if nbytes > cap(b) {
    nbytes = cap(b)
  }
  if nbytes > len(b) {
    b = b[:nbytes]
  }
}
```

Reply:
Do you mean:

```
if nbytes > len(b) {
  if nbytes > cap(b) {
    nbytes = cap(b)
  }
  if nbytes > len(b) {
    b = b[:nbytes]
  }
}
```

Reply:
Thanks for checking back.  Now that I see this code, I prefer what you originally had.  Apologies!